### PR TITLE
Improve getBlocks request handling

### PR DESCRIPTION
**Pull request by chimp1984**

### CODE REVIEW

**Review by ghubstan — COMMENTED** [review]
utACK

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]:
```suggestion
        } else if (blockDownloadListener == null) {
            blockDownloadListener = (observable, oldValue, newValue) -> {
                if ((double) newValue == 1) {
                    setupWalletBestBlockListener();
                    UserThread.execute(() -> walletsSetup.downloadPercentageProperty().removeListener(blockDownloadListener));
                }
            };
            walletsSetup.downloadPercentageProperty().addListener(blockDownloadListener);
        }
```

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: Why error log level?

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: Timeout unit should be sec, is this a test artefact together with the error log level?

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: That's quite a bit of logging but perhaps ok since it's once per received block

File: core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java]:
```suggestion
                                "Might be caused by a previous networkNode.sendMessage.onFailure.");
```

File: core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java:165]:
```suggestion
        if (!(networkEnvelope instanceof GetBlocksResponse)) return; // Also need to remove other brace
```

File: core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java
Comment (sqrrm) [inline core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java]: Doesn't timeoutTimer need to be stopped here like before?
```
        stopTimeoutTimer();
```

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: I will make a commit with the suggestion. Have better control if I do it in the IDE...

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: Ups was from dev testing

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: Ups, another dev setup leftover ;-(

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: We can lower loglevel once we see that the feature works more reliable. Prefer to keep it now.

File: core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java]: Found more like that. Will fix in new commit.

File: core/src/main/java/bisq/core/dao/node/lite/LiteNode.java
Comment (sqrrm) [reply core/src/main/java/bisq/core/dao/node/lite/LiteNode.java]: Yes, I only make the code suggestion here as a way to explain more precisely what I mean. Best do it on your own to be sure it's correct.

File: core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java]: cleanup calls stopTimeoutTimer. But will consolidate cancel/stop/cleanup

File: core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java
Comment (chimp1984) [reply core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java:165]: I prefer to not return early here as this style is commonly used in many places and often there are several if else branches. It is more like a dispatcher or switch case style (if switch would be more flexible in java). I prefer to use the early return for validation use cases but here its more to select the msg we are interested in.

Comment (sqrrm) [reply core/src/main/java/bisq/core/dao/node/lite/network/RequestBlocksHandler.java:165]: That's a good point. Might be cleaner to make a `handleGetBlocksResponse(GetBlocksResponse r)` and other respective dispatch handlers like is done in the trade protocol for these cases.

**Comment by ripcurlx** [comment]
@sqrrm Is there anything else you want @chimp1984 to address?

**Comment by sqrrm** [comment]
@ripcurlx I have nothing more but in private @chimp1984 mentioned he might want to add some more so I'll wait for that to merge.

**Comment by chimp1984** [comment]
@ripcurlx I would like to do myself another review before merging it.

**Review by ManfredKarrer — APPROVED** [review]
utACK

**Review by sqrrm — APPROVED** [review]
utACK

**Pull request merged**
